### PR TITLE
Add ioredis to problematic-modules.md

### PR DESCRIPTION
**Pull request by cupofjoakim**

### CODE REVIEW

**Review by Qard — APPROVED** [review]
Hmm...this doc really needs a review, actually. A few things in there I'm fairly certain are working at this point, and AsyncHooks now exists in all supported Node.js versions so the leading paragraphs are inaccurate. 🤔 

Anyway, this change LGTM.

**Review by RafaelGSS — APPROVED** [review]
LGTM.

**Comment by mmarchini** [comment]
Woah I had no idea we have had that list. that list seems pretty, uhm, problematic (😅)? At least with the wording used. Honestly I don't think we should have a list unless we're committed to work with package authors to improve the situation. And _if_ we decide to have a list, it should be triaged often and we should have links/examples showing how the library is problematic/breaks async continuity.

**Comment by mmarchini** [comment]
Added to the agenda so we can discuss what to do with this list

**Comment by RafaelGSS** [comment]
> Honestly I don't think we should have a list unless we're committed to work with package authors to improve the.

Indeed +1.

**Comment by cupofjoakim** [comment]
@mmarchini It's nice that you added it to the agenda, especially since the comments gathered here is more about the existence of the doc than the change. 

As a user in the enterprise world with hard requirements that in turn forces us to use async_hooks, there's a huge value in having the document as we often won't see the issues before we're in production. Too bad it's not being actively maintained though - I guess it's better not to have it than having it unmaintained.

**Comment by Flarna** [comment]
I agree that we should remove the list.
By the way, if we keep it we should also add Nodejs 12 and 10...

File: tracing/AsyncHooks/problematic-modules.md [hunk 1]
```diff
@@ -18,3 +18,4 @@ Module | Embedder API | Note
 [generic-pool](https://github.com/coopernurse/node-pool) | No | Used by a lot of database modules like for instance "pg"
 [mongoose](https://github.com/Automattic/mongoose/) | [#5929](https://github.com/Automattic/mongoose/issues/5929) | Most popular MongoDB ORM/ODM that implements its own user-land queueing.
 [graphql-subscriptions](https://github.com/apollographql/graphql-subscriptions/blob/26ed503566ecfab086e7530f0daf12b60ee3049c/src/pubsub-async-iterator.ts) | No | De-facto standard for working with GraphQL subscriptions. Custom queueing via an async iterator written in TypeScript and compiled down to ES5.
+[ioredis](https://github.com/luin/ioredis) | No | The only commonly used redis client with support for sentinels
```
Comment (mhdawson) [inline tracing/AsyncHooks/problematic-modules.md:21]: As a question is the other widely used redis package problematic (https://www.npmjs.com/package/redis) as well ?

**Comment by mmarchini** [comment]
As decided in the meeting we'll remove this file and create separate issues for each module that breaks async continuity, so we can work with module maintainers to fix it.

**Comment by cupofjoakim** [comment]
@mmarchini I understand! What does that mean for this PR? Should I close it?

File: tracing/AsyncHooks/problematic-modules.md (hunk 1 above)
Comment (cupofjoakim) [reply tracing/AsyncHooks/problematic-modules.md:21]: @mhdawson I have no idea actually. It doesn't support sentinels, so I can't really try it out at work.

**Comment by mmarchini** [comment]
Keep it open, we can close it once the issues are created (will try to get it done this week)

**Comment by mmarchini** [comment]
Issues created for all modules in the file (plus this one): https://github.com/nodejs/diagnostics/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Aasync-continuity

I'll close this PR and remove the file now. @cupofjoakim thank you for bringing it to our attention!

**Pull request closed without merging**